Entities/HDFirearm.cpp

```diff
@@ -305,14 +312,38 @@ int HDFirearm::Save(Writer &writer) const
 
 void HDFirearm::Destroy(bool notInherited)
 {
-    m_PreFireSound.Stop();
-    m_FireSound.Stop();
-    m_FireEchoSound.Stop();
-    m_ActiveSound.Stop();
-    m_DeactivationSound.Stop();
-    m_EmptySound.Stop();
-    m_ReloadStartSound.Stop();
-    m_ReloadEndSound.Stop();
+	if (m_PreFireSound) {
+		m_PreFireSound->Stop();
+		delete m_PreFireSound;
```

Comment:
I'm not picky on this. but you don't need the delete lines in the if checks, and I don't believe you did it that way in other files. Feel free to leave it as-is, or change it for consistency, I'm cool either way :)

Reply:
No worries, Changed to match.